qgreenland/config/layers/Glaciology/Ice sheet velocity/GrIMP/grimp.py

```diff
@@ -0,0 +1,179 @@
+* A temporal offset parameter (dT) that reports the difference in days between
+  the date of each velocity estimate and the midpoint date of the corresponding
+  measurement period.
+* Shapefile that indicates the source of the image pairs (SAR or Landsat 8) used
+  to produce the mosaic.
+"""
+
+
+_layer_params = {
+    "vv": {
+        "description": "Ice sheet velocity magnitude (vv) in meters per year for 2021."
+        + _description_common,
+        "style": "grimp_velocity_magnitude",
+        "title": "Annual ice sheet velocity magnitude 2021 (200m)",
+    },
+    "vx": {
+        "description": "Ice sheet velocity x component (vy) in meters per year for 2021."
+        + _description_common,
+        "style": "grimp_velocity_component",
+        "title": "Annual ice sheet velocity x component 2021 (200m)",
+    },
+    "vy": {
+        "description": "Ice sheet velocity y component (vy) in meters per year for 2021."
+        + _description_common,
+        "style": "grimp_velocity_component",
+        "title": "Annual ice sheet velocity y component 2021 (200m)",
+    },
+}
+
+annual_grimp_layers = [
+    make_layer(
+        variable=variable,
+        layer_id=f"grimp_annual_{variable}_2021",
+        style=layer_params["style"],
+        description=layer_params["description"],
+        title=layer_params["title"],
+    )
+    for variable, layer_params in _layer_params.items()
+]
+
+grimp_vector_layer = Layer(
+    id="grimp_annual_vectors_2021",
+    title="Annual ice sheet velocity vectors 2021 (1.5km)",
+    description=(
+        """Vector representation of ice sheet velocity in meters per year for 2021."""
+    ),
+    style="grimp_vectors",
+    input=LayerInput(
+        dataset=annual_dataset,
+        asset=annual_dataset.assets["only"],
+    ),
+    steps=[
+        # Now merge the variables into a 3-band .tif file.
+        CommandStep(
+            args=[
+                "gdal_merge.py",
+                "-a_nodata",
+                "-2e+09",
+                "-separate",
+                "-o",
+                "{output_dir}/merged.tif",
+                "{input_dir}/*vx*.tif",
+                "{input_dir}/*vy*.tif",
+            ],
+        ),
+        # Downsample to 1.5km
+        CommandStep(
+            args=[
+                "gdalwarp",
+                "-tr",
+                "1500 1500",
+                "{input_dir}/merged.tif",
+                "{output_dir}/downsampled.tif",
+            ],
+        ),
+        # Next, convert the .tif into a csv file.
+        CommandStep(
+            args=[
+                "gdal2xyz.py",
+                "-skipnodata",
+                "-csv",
+                "-allbands",
+                "{input_dir}/downsampled.tif",
+                "{output_dir}/as_xyz.xyz",
+                "&&",
+                'echo "x,y,vx,vy" > {output_dir}/data_with_header.csv',
+                "&&",
+                "cat {output_dir}/as_xyz.xyz >> {output_dir}/data_with_header.csv",
```

Comment:
We could do this with `sed` to avoid the intermediate file.

```bash
sed -i '1s/^/x,y,vx,vy\n/' {output_dir}/as_xyz.xyz
```

`-i`: edit in place
`1s`: edit the first line